src/Homework06/Main2.java

```diff
@@ -0,0 +1,28 @@
+package Homework06;
+
+public class Main2 {
+
+    /**
+     * процедура принимающая на вход массив из целых чисел, и выводящая в консоль значимые числа влево, а нули вправо.
+     * @param array - входящий массив целых чисел.
+     */
+    public  static void autoSorting(int[] array){
+        int[] outputArray = new int[array.length];
+        int indexOutputArray = 0;
+        for (int i = indexOutputArray; i < array.length; i++) {
+            if(array[i] != 0){
+                outputArray[indexOutputArray] = array[i];
+                indexOutputArray++;
+            }
+
+        }
+        for (int j : outputArray) {
+            System.out.print(j + ", ");
+        }
```

Comment:
Некритично, но здесь у тебя не счетчик а элемент массива, поэтому не жалей букв на осмысленные названия.

```suggestion
        for (int number : outputArray) {
            System.out.print(number + ", ");
        }
```
Так прям читается: для числа в массиве - напечатай число.
Еще лучше System.out.println(Arrays.toString(outputArray)) чтобы не получать разделительные символы в конце строки (не будет ", " после замыкающей цифры).

Reply:
Согласен. Так лучше. Исправил.